### PR TITLE
BHoM: design the representation system

### LINKED ISSUE #1180: BHoM: design the representation system

### PR DESCRIPTION
**Pull request by alelom**
### Issues addressed by this PR

Closes #1180

### CODE REVIEW

File: Geometry_oM/Representation/GeometricalRepresentationOptions.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using System.ComponentModel;
+using System.Drawing;
+
+namespace BH.oM.Geometry
+{
+    public interface IGeometricalRepresentationOptions : IRepresentationOptions
```
[the remark this answers is not in the document]
Comment (alelom) [reply Geometry_oM/Representation/GeometricalRepresentationOptions.cs:29]: Agreed the need of a better spot - was placed there initially as prototype. Although considerations are needed for the reachability of the class from everywhere, possibly same dependants as Geometry. It means lowering the `Graphics` in the dependancy chain. Will review asap.

[the remark this answers is not in the document]
Comment (al-fisher) [reply Geometry_oM/Representation/GeometricalRepresentationOptions.cs:29]: Ah OK - I see
👍

File: Geometry_oM/Representation/GeometricalRepresentation.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+using System.ComponentModel;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.oM.Geometry
+{
+    [Description("A geometrical object that is the representation of another object.")]
+    public class GeometricalRepresentation : IRepresentation
```
Comment (al-fisher) [inline Geometry_oM/Representation/GeometricalRepresentation.cs:13]: With potential to align with Existing `RenderMesh` object in the `Graphics` namespace thinking we could name this object simply `RenderGeometry` or perhaps `DisplayGeometry` it is geometry with Colour info to help display it.

What do @rolyhudson @alelom @IsakNaslundBh think?

File: Geometry_oM/Representation/TextRepresentation.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.oM.Geometry
+{
+    public class TextRepresentation : IRepresentation
```
Comment (al-fisher) [inline Geometry_oM/Representation/TextRepresentation.cs:12]: Same as comment above - perhaps `RenderText` or `DisplayText` based on prefered option above

File: Geometry_oM/Representation/GeometricalRepresentation.cs (hunk 2 above)
Comment (IsakNaslundBh) [reply Geometry_oM/Representation/GeometricalRepresentation.cs:13]: +1 @al-fisher

Comment (rolyhudson) [reply Geometry_oM/Representation/GeometricalRepresentation.cs:13]: I quite like the use of Render following a formal definition of the term:

> [to cause someone or something to be in a particular state](https://dictionary.cambridge.org/dictionary/english/render)

`IRender` in that sense may be more appropriate than `IRepresentation`

**Comment by FraserGreenroyd** [comment]
Discussed with @alelom today concluded this PR can be closed as this will be replaced by further work around the Geometry Hash and representation in 2023.

**Pull request closed without merging**